File: C/SEM I/Practical/Done/47.c
/* Write a program to convert a decimal number into a binary number */

#include <stdio.h>
#include <conio.h>
#include <stdlib.h>
#include <math.h>

typedef int intArr;

int count = 0;

int main(void) {
	int rep = 1;

	while (rep == 1) {
		int dec, temp;
		intArr bin[50];
		int i, j;

		printf("Enter the decimal number: ");
		scanf("%d", & dec);

		temp = dec;

		for (i = 0; temp > 0; i++) {
			bin[i] = temp % 2;
			temp /= 2;
		}

		printf("Decimal %d = Binary ", dec);

		for (j = i - 1; j >= 0; j--) {
			printf("%d", bin[j]);
		}

		printf("\nRepeat Program?: ");
		scanf("%d", & rep);

		count++;

		if (count >= 50) {
			printf("Abnormal END\n");

			getch();
			exit(0);
		}

	}

	printf("Normal END\n");

	getch();
	return 0;
}

File: Python/practical/47.py
# Program to convert decimal to binary

import math

def decimalToBinary(decimal):
    binary = []
    
    while decimal > 0:
        binary.append(decimal % 2)
        decimal = math.floor(decimal / 2)
        
    binary.reverse()
    
    return binary

# decimal = int(input("Enter a decimal number: "))
decimal = 5

binary = decimalToBinary(decimal)

print(f"Decimal {decimal} = Binary {''.join(map(str, binary))}")